mp/term.py

```diff
@@ -65,10 +69,132 @@ def set_rx_encoding(self, enc):
         def set_tx_encoding(self, enc):
             pass
 
-
 else:
 
     # see if we could use the new Miniterm implementation
-    from serial.tools.miniterm import Miniterm
-
-    Term = Miniterm
+    from serial.tools.miniterm import Miniterm, ConsoleBase, unichr
+    import os
+    if os.name == 'nt':  # noqa
+        import codecs
+        import sys
+        import msvcrt
+        import ctypes
+        import platform
+
+        class Out(object):
+            """file-like wrapper that uses os.write"""
+
+            def __init__(self, fd):
+                self.fd = fd
+
+            def flush(self):
+                pass
+
+            def write(self, s):
+                os.write(self.fd, s)
+
+        class Console(ConsoleBase):
+            fncodes = {
+                ';': '\1bOP',  # F1
+                '<': '\1bOQ',  # F2
+                '=': '\1bOR',  # F3
+                '>': '\1bOS',  # F4
+                '?': '\1b[15~',  # F5
+                '@': '\1b[17~',  # F6
+                'A': '\1b[18~',  # F7
+                'B': '\1b[19~',  # F8
+                'C': '\1b[20~',  # F9
+                'D': '\1b[21~',  # F10
+            }
+            navcodes = {
+                'H': '\x1b[A',  # UP
+                'P': '\x1b[B',  # DOWN
+                'K': '\x1b[D',  # LEFT
+                'M': '\x1b[C',  # RIGHT
+                'G': '\x1b[H',  # HOME
+                'O': '\x1b[F',  # END
+                'R': '\x1b[2~',  # INSERT
+                'S': '\x1b[3~',  # DELETE
+                'I': '\x1b[5~',  # PGUP
+                'Q': '\x1b[6~',  # PGDN        
+            }
+            
+            def __init__(self):
+                super(Console, self).__init__()
+                self._saved_ocp = ctypes.windll.kernel32.GetConsoleOutputCP()
+                self._saved_icp = ctypes.windll.kernel32.GetConsoleCP()
+                ctypes.windll.kernel32.SetConsoleOutputCP(65001)
+                ctypes.windll.kernel32.SetConsoleCP(65001)
+                # ANSI handling available through SetConsoleMode since Windows 10 v1511 
+                # https://en.wikipedia.org/wiki/ANSI_escape_code#cite_note-win10th2-1
+                # if platform.release() == '10' and int(platform.version().split('.')[2]) > 10586:
+                ENABLE_VIRTUAL_TERMINAL_PROCESSING = 0x0004
+                import ctypes.wintypes as wintypes
+                if not hasattr(wintypes, 'LPDWORD'): # PY2
+                    wintypes.LPDWORD = ctypes.POINTER(wintypes.DWORD)
+                SetConsoleMode = ctypes.windll.kernel32.SetConsoleMode
```

Comment:
This is a lot of Windows-specific code that I have no experience in, so I will take your word for it.

Reply:
> This is looking good, I don't have Windows so I can't test this and I don't have experience with it, but if it works for you, that's fine with me. It doesn't seem to do anything if the OS is anything other than Windows, so it shouldn't break anything else either.



> This is looking good, I don't have Windows so I can't test this and I don't have experience with it, but if it works for you, that's fine with me. It doesn't seem to do anything if the OS is anything other than Windows, so it shouldn't break anything else either.

This can rest assured that I have in WIN7 and WIN10 to provide my customer groups to use.

It has also been tested on Linux and Mac.